pom.xml

```diff
@@ -683,6 +683,10 @@
                                     <pattern>org.apache.commons</pattern>
                                     <shadedPattern>${shadeBase}.org.apache.commons</shadedPattern>
                                 </relocation>
+                                <relocation>
+                                    <pattern>org.apache.orc</pattern>
+                                    <shadedPattern>${shadeBase}.org.apache.orc</shadedPattern>
```

Comment:
We reference this directly in Trino, so we should change the shade base to `io.trino.hive.orc`. And move the relocation to the top, since it's different than the others.

Reply:
In Trino we could use a separately loaded (newer) version of ORC. Would that work? Would that be instead of putting this in a nice package?

Reply:
I tried using separated version in trino but it generated a lot of problems. Like maven started complaining that `hive-storage` is used(orc uses it) but not declared. Same about `hadoop-common`. 
I may be wrong but I thought that we have this `trino-hive-apache` to prevent us from having a mess with hive/hadoop dependencies..

Reply:
That works for the production `OrcConf` usage in `trino-hive`, since it's just setting properties. Though it's cleaner to use the shaded version, since it's being consumed by that code.

For `trino-raptor`, we need the shaded version since it's interacting with the Hive writer code. We could eliminate this by making Raptor use the Trino ORC writer code (I have this in a Raptor v2 branch if we want to cherry pick it).

Most of the test usages need to use the shaded version, since they are interacting with the Hive writer code.

Reply:
We can't change it for the Hive connector, but we should be able to add it as a dependency for the Iceberg connector.

Reply:
So Iceberg connector would effectively use 2 ORCs?
via Iceberg library it would use ORC 1.7.x
and via trino-orc it would continue to use ORC 1.5.x, right?

Reply:
> So Iceberg connector would effectively use 2 ORCs? via Iceberg library it would use ORC 1.7.x and via trino-orc it would continue to use ORC 1.5.x, right?

I didn't consider this but it looks like it is the case. I hate it but I don't have any idea what else could be done. 
Removing orc from trno-hive-apache won't work also because we would need to declare dependencies I mentioned above in trino..

Reply:
`trino-orc` only uses Apache ORC for testing. Iceberg would have the code via `hive-apache` but would never use it.

Reply:
what does `trino-hive` uses ORC for (except tests)? besides creating empty bucket files?